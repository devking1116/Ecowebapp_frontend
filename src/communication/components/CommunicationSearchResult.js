/* @flow */
import React from 'react';
import reactCSS from 'reactcss';
import { Icon } from 'semantic-ui-react';

export type Props = {
  title: string,
  author: string,
  date: string,
  status?: 'active' | 'in-active',
  isExpanded?: bool,
};

export default function CommunicationSearchResult(props: Props): React.Element<*> {
  const { title, author, date, status, isExpanded, ...otherProps } = props;
  const styles = reactCSS({
    'default': {
      wrapper: {
        display: 'flex',
        height: 65,
      },
      icon: {
        display: 'flex',
        flexDirection: 'column',
        justifyContent: 'space-around',
        width: 15,
        marginRight: 20,
      },
      info: {
        display: 'flex',
        flexDirection: 'column',
        justifyContent: 'space-around',
      },
      title: {
        fontWeight: 'bold',

        //TODO, move to theme
        color: '#468dc8',
      },
      date: {
        //TODO, move to theme
        color: '#468dc8',
      },
      author: {
        //TODO, move to theme
        color: '#6da94f',
      },
    }
  });

  return (
    <div style={styles.wrapper} {...otherProps}>
      <div style={styles.icon}>
      {status === 'active'?
        <Icon name="circle" color="green" />:
        null
      }
      {status === 'in-active'?
        <Icon name="circle thin" color="grey" />:
        null
      }
      </div>
      <div style={styles.info}>
        <span style={styles.title}>{title}</span>
        <span style={styles.date}>{date}</span>
        <span style={styles.author}>{author}</span>
      </div>
    </div>
  );
}
