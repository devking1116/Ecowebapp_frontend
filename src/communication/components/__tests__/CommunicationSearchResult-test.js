import React from 'react';
import renderer from 'react-test-renderer';

import CommunicationSearchResult from '../CommunicationSearchResult';

test('it displays the given title, author and date.', () => {
  const component = renderer.create(
    <CommunicationSearchResult title="Test title" author="Test author" date="01/02/2016" />
  );
  const tree = component.toJSON();

  expect(tree).toMatchSnapshot();
});

test('it dosen\'t display the icon if no status is given.', () => {
  const component = renderer.create(
    <CommunicationSearchResult />
  );
  const tree = component.toJSON();

  expect(tree).toMatchSnapshot();
});

test('it display the green icon if status is "active" and grey if "inactive".', () => {
  const componentActive = renderer.create(
    <CommunicationSearchResult status="active" />
  );
  const componentInactive = renderer.create(
    <CommunicationSearchResult status="inactive" />
  );
  const treeActive = componentActive.toJSON();
  const treeInactive = componentInactive.toJSON();

  expect(treeInactive).toMatchSnapshot();
});
