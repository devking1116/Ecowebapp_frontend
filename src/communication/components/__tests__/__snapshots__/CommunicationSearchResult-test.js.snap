exports[`test it display the green icon if status is "active" and grey if "inactive". 1`] = `
<div
  style={
    Object {
      "display": "flex",
      "height": 65,
    }
  }>
  <div
    style={
      Object {
        "WebkitJustifyContent": "space-around",
        "display": "flex",
        "flexDirection": "column",
        "justifyContent": "space-around",
        "marginRight": 20,
        "width": 15,
      }
    } />
  <div
    style={
      Object {
        "WebkitJustifyContent": "space-around",
        "display": "flex",
        "flexDirection": "column",
        "justifyContent": "space-around",
      }
    }>
    <span
      style={
        Object {
          "color": "#468dc8",
          "fontWeight": "bold",
        }
      } />
    <span
      style={
        Object {
          "color": "#468dc8",
        }
      } />
    <span
      style={
        Object {
          "color": "#6da94f",
        }
      } />
  </div>
</div>
`;

exports[`test it displays the given title, author and date. 1`] = `
<div
  style={
    Object {
      "display": "flex",
      "height": 65,
    }
  }>
  <div
    style={
      Object {
        "WebkitJustifyContent": "space-around",
        "display": "flex",
        "flexDirection": "column",
        "justifyContent": "space-around",
        "marginRight": 20,
        "width": 15,
      }
    } />
  <div
    style={
      Object {
        "WebkitJustifyContent": "space-around",
        "display": "flex",
        "flexDirection": "column",
        "justifyContent": "space-around",
      }
    }>
    <span
      style={
        Object {
          "color": "#468dc8",
          "fontWeight": "bold",
        }
      }>
      Test title
    </span>
    <span
      style={
        Object {
          "color": "#468dc8",
        }
      }>
      01/02/2016
    </span>
    <span
      style={
        Object {
          "color": "#6da94f",
        }
      }>
      Test author
    </span>
  </div>
</div>
`;

exports[`test it dosen't display the icon if no status is given. 1`] = `
<div
  style={
    Object {
      "display": "flex",
      "height": 65,
    }
  }>
  <div
    style={
      Object {
        "WebkitJustifyContent": "space-around",
        "display": "flex",
        "flexDirection": "column",
        "justifyContent": "space-around",
        "marginRight": 20,
        "width": 15,
      }
    } />
  <div
    style={
      Object {
        "WebkitJustifyContent": "space-around",
        "display": "flex",
        "flexDirection": "column",
        "justifyContent": "space-around",
      }
    }>
    <span
      style={
        Object {
          "color": "#468dc8",
          "fontWeight": "bold",
        }
      } />
    <span
      style={
        Object {
          "color": "#468dc8",
        }
      } />
    <span
      style={
        Object {
          "color": "#6da94f",
        }
      } />
  </div>
</div>
`;
